src/main/java/io/stargate/sgv2/jsonapi/config/DocumentLimitsConfig.java

```diff
@@ -1,10 +1,15 @@
 package io.stargate.sgv2.jsonapi.config;
 
+import io.quarkus.runtime.annotations.StaticInitSafe;
 import io.smallrye.config.ConfigMapping;
 import io.smallrye.config.WithDefault;
 import jakarta.validation.constraints.Positive;
 
-/** Configuration Object that defines limits on Documents managed by JSON API. */
+/**
+ * Configuration Object that defines limits on Documents managed by JSON API. Needed early for
+ * providers so has to be declared as {@link StaticInitSafe}.
+ */
+@StaticInitSafe
```

Comment:
Injection into ObjectMapper provider failed without adding this annotation.

Reply:
Are you sure this is not preventing the values to be changed during the runtime?